backend/src/main/java/ai/verta/modeldb/experiment/ExperimentDAORdbImpl.java

```diff
@@ -720,7 +720,9 @@ public Experiment logExperimentCodeVersion(String experimentId, CodeVersion upda
               RdbmsUtils.generateArtifactEntity(
                   experimentEntity,
                   ModelDBConstants.CODE_ARCHIVE,
-                  updatedCodeVersion.getCodeArchive()));
+                  updatedCodeVersion.getCodeArchive(),
+                  CodeVersionEntity.class.getSimpleName(),
+                  Optional.empty()));
```

Comment:
This seems wrong to me as it's prone to collisions. Why would we ever not have the entity id?

Reply:
We have code_version in `gitsnapshot` and we have used same `artifact` entity for them and that code_version entity have auto generated id we can not set it in our entity before insertion that's the issue so i have make it optional

Reply:
It doesn't solve the collision problem. It just confirms that collision _will_ be a problem. We need to have an id before saving to S3.

Reply:
Used UUID for uniqueness as per discussion and removed optional from codebase